agent/xds/clusters.go

```diff
@@ -385,6 +385,16 @@ func (s *ResourceGenerator) injectGatewayServiceAddons(cfgSnap *proxycfg.ConfigS
 			if mapping.SNI != "" {
 				tlsContext.Sni = mapping.SNI
 			}
+			err := injectSANMatcher(tlsContext.CommonTlsContext, connect.SpiffeIDService{
```

Comment:
I think that this should instead be using the SNI when available. The issue with using a consul SPIFFE ID is that terminating gateways route to external services, to which we don't provision certificates. 

Since specifying the SNI in a terminating gateway config entry is optional, there are a couple more things we could do:
* Add a strong recommendation to the config entry docs that the SNI should be set when using TLS.
* Log a warning on config write, when any of: CAFile, CertFile, or KeyFile are specified in the config entry, but not SNI.

Reply:
That makes sense - I guess it would be strange to expect people to fetch connect certs for their external service. So all we can really do here is update the docs and maybe add that log message when creating/updating a terminating gateway config entry.

Reply:
@picatz Are you on board with just a doc change for this item?

Reply:
It wouldn't just be a docs change, we would also start validating SANs when an SNI is provided in the terminating-gateway config entry.

The docs change is about recommending that people provide an SNI, so that we can do that upstream validation